core/src/main/java/io/atomix/core/value/impl/AbstractAtomicValueService.java

```diff
@@ -106,4 +107,14 @@ public void addListener() {
   public void removeListener() {
     listeners.remove(getCurrentSession().sessionId());
   }
+
+  @Override
+  protected void onExpire(Session session) {
+    listeners.remove(session.sessionId());
```

Comment:
I assume if a client times out and then reconnects the listener will be re-registered?

Reply:
yep